File: api-backoffice/elenco-ordini/java/elenco-ordini.java
                                            
// Elenco Ordini

import java.security.MessageDigest;
import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClientBuilder;
import org.apache.http.util.EntityUtils;
import java.text.SimpleDateFormat;
import java.util.Date;
import org.json.JSONObject;

public class elencoOrdini {

    public static void main(String[] a) throws Exception {

        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
        Date date = new Date();

		// URL + URI
        String requestUrl = "https://int-ecommerce.nexi.it/" + "ecomm/api/bo/reportOrdini";

        // Parametri per calcolo MAC
        String apiKey = "ALIAS_WEB_00002830"; // Alias fornito da Nexi
        String chiaveSegreta = "9R6JMIHU9SDUBFH62IUXSQ7VOSD17VMP"; // Chiave segreta fornita da Nexi
        String codTrans = ""; // Vuoto per tutte le transazioni altrimenti cerca la transazione inserita
        String periodo = "01/07/2019 - 30/07/2019"; // gg/mm/aaaa - gg/mm/aaaa
        String canale = "All"; // All || MyBank || CartaCredito || PayPal
        String stato[] = {"Autorizzato"}; // Transazioni Autorizzate
        String timeStamp = "" + System.currentTimeMillis();        
		
        String stringaMac = "apiKey=" + apiKey
                + "codiceTransazione=" + codTrans
                + "periodo=" + periodo
                + "canale=" + canale
                + "timeStamp=" + timeStamp
                + chiaveSegreta;

        // Calcolo MAC
        String macCalculated = hashMac(stringaMac);

        // Parametri di invio
        JSONObject json = new JSONObject();
        json.put("apiKey", apiKey);
        json.put("codiceTransazione", codTrans);
        json.put("periodo", periodo);
        json.put("canale", canale);
        json.put("stato", stato);
        json.put("timeStamp", timeStamp);
        json.put("mac", macCalculated);

        // Chiamata API
        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
        HttpPost request = new HttpPost(requestUrl);
        StringEntity params = new StringEntity(json.toString());
        request.addHeader("content-type", "application/json");
        request.setEntity(params);
        HttpResponse response = httpClient.execute(request);
        HttpEntity entity = response.getEntity();
        String responseString = EntityUtils.toString(entity, "UTF-8");

        // Parametri di ritorno
        JSONObject responseObj = new JSONObject(responseString);
        
        System.out.print(responseObj.toString());

    }

    public static String hashMac(String stringaMac) throws Exception {
        MessageDigest digest = MessageDigest.getInstance("SHA-1");
        byte[] in = digest.digest(stringaMac.getBytes("UTF-8"));

        final StringBuilder builder = new StringBuilder();
        for (byte b : in) {
            builder.append(String.format("%02x", b));
        }
        return builder.toString();
    }
}
                    
                

File: api-backoffice/elenco-ordini/php/elenco-ordini.php
                                            
<?php

// Elenco Ordini

$connection = curl_init();

if ($connection) {
    
    $requestURL = "https://int-ecommerce.nexi.it/"; // URL
    $requestURI = "ecomm/api/bo/reportOrdini"; // URI
    
    // Parametri per calcolo MAC
    $apiKey = "<ALIAS>"; // Alias fornito da Nexi
    $chiaveSegreta = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Chiave segreta fornita da Nexi
    $codiceTransazione = ""; // Vuoto per tutte le transazioni altrimenti cerca la transazione inserita
    $periodo = "01/01/2017 - 31/12/2017"; // gg/mm/aaaa - gg/mm/aaaa 
    $canale = "All"; // All || MyBank || CartaCredito || PayPal
    $stato = array("Autorizzato", "Negato", "annullato"); // Transazioni Autorizzate o Negate o annullate
    $timeStamp = (time()) * 1000;

    // Calcolo MAC
    $mac = sha1('apiKey=' . $apiKey . 'codiceTransazione=' . $codiceTransazione . 'periodo=' . $periodo . 'canale=' . $canale . "timeStamp=" . $timeStamp . $chiaveSegreta);

    // Parametri
    $parametri = array(
        'apiKey' => $apiKey,
        'codiceTransazione' => $codiceTransazione,
        'periodo' => $periodo,
        'canale' => $canale,
        'stato' => $stato,
        'timeStamp' => $timeStamp,
        'mac' => $mac
    );

    curl_setopt_array($connection, array(
        CURLOPT_URL => $requestURL . $requestURI,
        CURLOPT_HTTPHEADER => array('Content-Type: application/json'),
        CURLOPT_POST => 1,
        CURLOPT_POSTFIELDS => json_encode($parametri),
        CURLOPT_RETURNTRANSFER => 1,
        CURLINFO_HEADER_OUT => true,
        CURLOPT_SSL_VERIFYPEER => 0
    ));

    $risposta = curl_exec($connection);

    curl_close($connection);

    // Decodifico risposta
    $json = json_decode($risposta, true);

    // Controllo JSON di risposta
    if (json_last_error() === JSON_ERROR_NONE) {

        $MACrisposta = sha1('esito=' . $json['esito'] . 'idOperazione=' . $json['idOperazione'] . 'timeStamp=' . $json['timeStamp'] . $chiaveSegreta);

        // Controllo MAC di risposta
        if ($json['mac'] == $MACrisposta) {

            // Controllo esito
            if ($json['esito'] == 'OK') {
                echo 'Operazione n. ' . $json['idOperazione'] . ' eseguita<br>';
                
                echo '<pre>';
                print_r($json['report']);
                echo '</pre>';
            } else {
                echo 'Operazione n. ' . $json['idOperazione'] . ' non eseguita. esito ' . $json['esito'] . '<br><br>' . json_encode($json['errore']);
            }
        } else {
            echo 'Errore nel calcolo del MAC di risposta';
        }
    } else {
        echo 'Errore nella lettura del JSON di risposta';
    }
} else {
    echo "Errore curl";
}
                    
                